From bdd9533639747cf7d2ce54fe84cf044a4dbc45d7 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:09:25 -0400
Subject: [usb] xhci: unified loggig of RESET_ON_RESUME

Message-id: <1410887463-20674-494-git-send-email-dzickus@redhat.com>
Patchwork-id: 95645
O-Subject: [RHEL7 PATCH 493/591] xhci: unified loggig of RESET_ON_RESUME
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 85f4e45b11e5d351789e09c259620d037246f6d8
Author: Oliver Neukum <oneukum@suse.de>
Date:   Wed May 14 14:00:23 2014 +0200

    xhci: unified loggig of RESET_ON_RESUME

    Either we log for all chips we set the quirk for or for
    none. This patch reports it for all chips.

    Signed-off-by: Oliver Neukum <oliver@neukum.org>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/host/xhci-pci.c b/drivers/usb/host/xhci-pci.c
index 35d4477..ffd119e 100644
--- a/drivers/usb/host/xhci-pci.c
+++ b/drivers/usb/host/xhci-pci.c
@@ -140,8 +140,6 @@ static void xhci_pci_quirks(struct device *dev, struct xhci_hcd *xhci)
  if (pdev->vendor == PCI_VENDOR_ID_ETRON &&
    pdev->device == PCI_DEVICE_ID_ASROCK_P67) {
   xhci->quirks |= XHCI_RESET_ON_RESUME;
-  xhci_dbg_trace(xhci, trace_xhci_dbg_quirks,
-    "QUIRK: Resetting on resume");
   xhci->quirks |= XHCI_TRUST_TX_LENGTH;
  }
  if (pdev->vendor == PCI_VENDOR_ID_RENESAS &&
@@ -149,6 +147,10 @@ static void xhci_pci_quirks(struct device *dev, struct xhci_hcd *xhci)
   xhci->quirks |= XHCI_RESET_ON_RESUME;
  if (pdev->vendor == PCI_VENDOR_ID_VIA)
   xhci->quirks |= XHCI_RESET_ON_RESUME;
+
+ if (xhci->quirks & XHCI_RESET_ON_RESUME)
+  xhci_dbg_trace(xhci, trace_xhci_dbg_quirks,
+    "QUIRK: Resetting on resume");
 }
 
 /* called during probe() after chip reset completes */
-- 
1.7.1